api/v1/helper.go

```diff
@@ -253,7 +253,12 @@ func NeedToUpdateSriov(ifaceSpec *Interface, ifaceStatus *InterfaceExt) bool {
 			for _, groupSpec := range ifaceSpec.VfGroups {
 				if IndexInRange(vfStatus.VfID, groupSpec.VfRange) {
 					ingroup = true
-					if groupSpec.DeviceType != consts.DeviceTypeNetDevice {
+					if vfStatus.Driver == "" {
+						log.V(2).Info("NeedToUpdateSriov(): Driver needs update - has no driver",
+							"desired", groupSpec.DeviceType)
+						return true
+					}
+					if groupSpec.DeviceType != "" && groupSpec.DeviceType != consts.DeviceTypeNetDevice {
```

Comment:
what is the reason for changing this check ?

Reply:
I think this check originally has a bug. This branch should run in case if device is not a network device. The problem is that empty  groupSpec.DeviceType also means network device.

Reply:
ack, i see.

empty DeviceType means the default which is "netdevice", mutating webhook sets this if not set by user.